### PR TITLE
Agent protocol client - Python (E2B-21)

### PR DESCRIPTION
**Pull request by jakubno**
Adding python client for interacting with agents

### CODE REVIEW

File: agent_client/python/examples/minimal.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+        task_id = response.task_id
+        response = await api_instance.execute_agent_task_step(task_id=task_id)
+        print("The response of AgentApi->execute_agent_task_step:\n")
+        print(response)
+        print("\n\n")
+
+        while (
+            step := await api_instance.execute_agent_task_step(task_id=task_id)
+        ) and step.is_last is False:
```
[the remark this answers is not in the document]
Comment (ValentaTomas) [reply agent_client/python/examples/minimal.py:36]: Yes, it is not shown in the SDK examples yet.

**Comment by ValentaTomas** [comment]
Also I think it is a good idea to either add installation steps for the `openapi-generator` or call it through package manager like `npx @openapitools/openapi-generator-cli generate` to skip the setup.

Some additional properties for the generator:
```
--additional-properties=disallowAdditionalPropertiesIfNotPresent=true \
--additional-properties=floatStrictType=false \
--additional-properties=allowStringInDateTimeParameters=false
```
were also pretty useful when I was using the SDK. If you think adding them is worth it we probably need to format the Makefile thought, because it will have like 350 chars :D

**Comment by jakubno** [comment]
> Also I think it is a good idea to either add installation steps for the `openapi-generator` or call it through package manager like `npx @openapitools/openapi-generator-cli generate` to skip the setup.
> 
> Some additional properties for the generator:
> 
> ```
> --additional-properties=disallowAdditionalPropertiesIfNotPresent=true \
> --additional-properties=floatStrictType=false \
> --additional-properties=allowStringInDateTimeParameters=false
> ```
> 
> were also pretty useful when I was using the SDK. If you think adding them is worth it we probably need to format the Makefile thought, because it will have like 350 chars :D

`disallowAdditionalPropertiesIfNotPresent` breaks the code, it's missing `from typing import Dict` and 
float strict type and allowStringInDateTimeParameters are not present for this [generator](https://github.com/OpenAPITools/openapi-generator/blob/v6.6.0/docs/generators/python-nextgen.md)

**Comment by ValentaTomas** [comment]
Nevermind then :D

**Comment by jakubno** [comment]
I will check the new version in beta, if they fixed it. Otherwise I will add an issue

File: agent_client/python/scripts/generate.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+set -e
+
+
+if ! command -v npx &> /dev/null
+then
+    echo "npx could not be found"
+    exit
+fi
+
+npm list -g | grep @openapitools/openapi-generator-cli || npm install -g @openapitools/openapi-generator-cli
```
Comment (ValentaTomas) [inline agent_client/python/scripts/generate.sh:12]: The command CLI should be callable via the `npx` (https://docs.npmjs.com/cli/v8/commands/npx) even without installation (it fetches the package if it is not installed locally or globally).

**Pull request merged**